tests/metrics/machine_learning/tensorflow.sh

```diff
@@ -89,59 +98,60 @@ function tensorflow_test() {
 	for pid in ${pids[*]}; do
 		wait "${pid}"
 	done
+}
 
-	info "All containers are running the workload..."
-
-	for i in "${containers[@]}"; do
-		check_file=$(sudo -E "${CTR_EXE}" t exec --exec-id "$(random_name)" "${i}" sh -c "${RESNET_CMD_FILE}")
-		retries="300"
-		for j in $(seq 1 "${retries}"); do
-			[ "${check_file}" -eq "1" ] && break
-			sleep 1
-		done
-	done
+function collect_results() {
+	WORKLOAD=${1}
+	[[ -z ${WORKLOAD} ]] && die "Container workload is missing"
 
-	info "Copy Alexnet Tensorflow test"
-	local pids=()
-	local j=0
-	for i in "${containers[@]}"; do
-		$(sudo -E "${CTR_EXE}" t exec -d --exec-id "$(random_name)" "${i}" sh -c "${CMD_ALEXNET}")&
-		pids["${j}"]=$!
-		((j++))
-	done
+	local tasks_running=("${containers[@]}")
+	local retries=${MAX_RETRIES}
 
-	# wait for all pids
-	for pid in ${pids[*]}; do
-		wait "${pid}"
-	done
+	while [ "${#tasks_running[@]}" -gt 0 ] && [ "${retries}" -gt 0 ]; do
+		for i in "${!tasks_running[@]}"; do
+			check_file=$(sudo -E "${CTR_EXE}" t exec --exec-id "$(random_name)" "${tasks_running[i]}" sh -c "${WORKLOAD}")
 
-	for i in "${containers[@]}"; do
-		check_file=$(sudo -E "${CTR_EXE}" t exec --exec-id "$(random_name)" "${i}" sh -c "${CMD_FILE}")
-		retries="300"
-		for j in $(seq 1 "${retries}"); do
-			[ "${check_file}" -eq "1" ] && break
-			sleep 1
+			# if the current task is done, remove the corresponding container from the active list
+			[ "${check_file}" -eq "1" ] && unset 'tasks_running[i]'
 		done
+		((retries--))
+		sleep 3
```

Comment:
This seems like a long time to sleep, why the increase?

Reply:
@cmaf thanks, the workload usually takes several minutes to complete (about 5 on my machine).
We added a "retries" variable as a timeout mechanism.